src/Interpreters/ActionsDAG.cpp

```diff
@@ -34,7 +34,7 @@ ActionsDAG::ActionsDAG(const NamesAndTypesList & inputs_)
         addInput(input.name, input.type, true);
 }
 
-ActionsDAG::ActionsDAG(const ColumnsWithTypeAndName & inputs_)
+ActionsDAG::ActionsDAG(const ColumnsWithTypeAndName & inputs_, QueryProcessingStage::Enum to_stage_)
```

Comment:
We definitely do not want to have `QueryProcessingStage` in ActionsDAG
At least, it can me special flag.

Reply:
Got it, will take a closer look, thanks!